relay/purger/purger.go

```diff
@@ -113,7 +114,9 @@ func (p *Purger) run() {
 	defer ticker.Stop()
 
 	var ctx context.Context
+	p.lock.Lock()
```

Comment:
still would override `p.cancel`, can we let only one goruotine running?

Reply:
in current concurrent model a `Purger` will be started in one `DM-worker` only. So we don't need to lock the whole `run` function.
We'd better to refine concurrent usage later

Reply:
if so, leave a `FIXME`

Reply:
added a note comment